File: Basic/tutorial-applestamp/contracts/src/test/java/com/tutorial/contracts/StateTests.java
package com.tutorial.contracts;

import com.tutorial.states.AppleStamp;
import com.tutorial.states.TemplateState;
import net.corda.core.identity.Party;
import org.junit.Test;

import static org.junit.Assert.assertSame;

public class StateTests {

    //Mock State test check for if the state has correct parameters type
    @Test
    public void hasFieldOfCorrectType() throws NoSuchFieldException {
        TemplateState.class.getDeclaredField("msg");
        assertSame(String.class, TemplateState.class.getDeclaredField("msg").getType());
    }

    @Test
    public void AppleStampStateHasFieldOfCorrectType() throws NoSuchFieldException {
        AppleStamp.class.getDeclaredField("stampDesc");
        assertSame(String.class, AppleStamp.class.getDeclaredField("stampDesc").getType());

        AppleStamp.class.getDeclaredField("issuer");
        assertSame(Party.class, AppleStamp.class.getDeclaredField("issuer").getType());

        AppleStamp.class.getDeclaredField("holder");
        assertSame(Party.class, AppleStamp.class.getDeclaredField("issuer").getType());
    }

}

File: java-samples/Basic/tutorial-jarsigning/contracts/src/test/kotlin/com/tutorial/contracts/StateTests.kt
package com.tutorial.contracts

import com.tutorial.states.AppleStamp
import com.tutorial.states.TemplateState
import net.corda.core.identity.Party
import org.junit.Test
import kotlin.test.assertEquals

class StateTests {
    @Test
    fun hasFieldOfCorrectType() {
        // Does the field exist?
        TemplateState::class.java.getDeclaredField("msg")
        // Is the field of the correct type?
        assertEquals(TemplateState::class.java.getDeclaredField("msg").type, String()::class.java)
    }

    @Test
    fun AppleStampStateHasFieldOfCorrectType(){
        AppleStamp::class.java.getDeclaredField("stampDesc")
        assert(AppleStamp::class.java.getDeclaredField("stampDesc").type == String::class.java)

        AppleStamp::class.java.getDeclaredField("issuer")
        assert(AppleStamp::class.java.getDeclaredField("issuer").type == Party::class.java)

        AppleStamp::class.java.getDeclaredField("holder")
        assert(AppleStamp::class.java.getDeclaredField("issuer").type == Party::class.java)
    }

}